openvdb_houdini/CMakeLists.txt

```diff
@@ -137,117 +219,257 @@ ADD_LIBRARY ( openvdb_houdini SHARED
+# @note _houdini_bin_dir is set from the HoudiniConfig.cmake module
+
+SET ( SESI_CACHE_TAG_DEFINE "" )
+IF ( OPENVDB_HOUDINI_CACHE_ONCE_SESI_TAG AND NOT OPENVDB_HOUDINI_DISABLE_SESI_TAG )
+  MESSAGE ( STATUS "Generating SESI Tag information for Houdini DSOs..." )
+  SET ( sesitag_input_file ${CMAKE_CURRENT_BINARY_DIR}/sesitag_input.cache/sesitag_input )
+
+  IF ( NOT EXISTS ${sesitag_input_file} )
+    STRING ( TIMESTAMP timestamp "%a %b %d %H:%M:%S %Y" )
+    SET ( tag_info "Generated on: ${timestamp}\n${CUSTOM_SESI_TAG_INFO}" )
+    FILE ( WRITE ${sesitag_input_file} "${tag_info}" )
+  ENDIF ()
+
+  SET ( sesitag_path "${_houdini_bin_dir}/sesitag${CMAKE_EXECUTABLE_SUFFIX}" )
+  EXECUTE_PROCESS (
+    COMMAND ${sesitag_path} -f ${sesitag_input_file}
+    RESULT_VARIABLE status_code
+    OUTPUT_VARIABLE SESI_CACHE_TAG_DEFINE
+  )
+
+  IF ( NOT status_code EQUAL 0 )
+    MESSAGE ( FATAL_ERROR "Error running sesitag: ${status_code}" )
+  ENDIF ()
+
+  STRING ( REPLACE "'" "" SESI_CACHE_TAG_DEFINE ${SESI_CACHE_TAG_DEFINE} )
+  STRING ( REPLACE "-D" "" SESI_CACHE_TAG_DEFINE ${SESI_CACHE_TAG_DEFINE} )
+  MESSAGE ( STATUS "Using cache SESI Tag data for Houdini DSOs: ${sesitag_input_file}" )
+ENDIF ()
+
+# Add all dsos as libraries and configure them for houdini
+
+MESSAGE ( STATUS "Configuring Houdini DSOs..." )
+
+FOREACH ( DSO_NAME ${OPENVDB_DSO_NAMES} )
+
+  # Create the lib
+
+  ADD_LIBRARY ( ${DSO_NAME} SHARED
+    houdini/${DSO_NAME}.cc
     )
 
-  target_compile_options(${SOP_NAME} PRIVATE -fext-numeric-literals)
+  # Link in deps
 
-  TARGET_LINK_LIBRARIES ( ${SOP_NAME}
+  TARGET_LINK_LIBRARIES ( ${DSO_NAME}
     openvdb_houdini
-    ${OPENVDB_SHARED_LIB}
+    ${OPENVDB_HOUDINI_DEPENDANT_LIBS}
     )
 
-  INSTALL ( TARGETS
-    ${SOP_NAME}
-    DESTINATION
-    ${OPENVDB_HFS_INSTALL_BASE_DIR}/dso
+  # Call houdini_configure_target to setup the sesi tag information. We set
+  # INSTDIR to the project build directory and configure the actuall install
+  # paths ourselves (otherwise cmake will configure the build to build directly
+  # to the install dir)
+
+  houdini_configure_target ( ${DSO_NAME}
+    INSTDIR ${PROJECT_BINARY_DIR}
+    PREFIX ""
+    TAGINFO ${CUSTOM_SESI_TAG_INFO}
+  )
+
+  # Add openvdb c flags
+
+  IF ( USE_LOG4CPLUS )
+    TARGET_COMPILE_DEFINITIONS ( ${DSO_NAME} PRIVATE "-DOPENVDB_USE_LOG4CPLUS" )
+  ENDIF ()
+
+  # If the sesi tag needs replacing or removing, manually configure the compile
+  # options for each DSO and replace the UT_DSO_TAGINFO define
+  # @todo ask for more options from sidefx for controlling tag timestamps
+
+  IF ( OPENVDB_HOUDINI_DISABLE_SESI_TAG OR SESI_CACHE_TAG_DEFINE )
+    GET_TARGET_PROPERTY ( DSO_DEFINITIONS ${DSO_NAME} COMPILE_DEFINITIONS )
+    STRING ( REGEX REPLACE "UT_DSO_TAGINFO=\".+\"" "${SESI_CACHE_TAG_DEFINE}" DSO_DEFINITIONS "${DSO_DEFINITIONS}")
+    SET_TARGET_PROPERTIES ( ${DSO_NAME}
+      PROPERTIES COMPILE_DEFINITIONS "${DSO_DEFINITIONS}"
     )
+  ENDIF ()
```

Comment:
@jmlait the above branch and lines 318-342 deal with the sesi tag stuff. We call `houdini_configure_target` in the Houdini CMake for each dso which appends `UT_DSO_TAGINFO` as a `COMPILE_DEFINITIONS` to the target, regenerating the timestamp. I've added two options, `OPENVDB_HOUDINI_DISABLE_SESI_TAG` which will remove the tag completely and `OPENVDB_HOUDINI_CACHE_ONCE_SESI_TAG` which will mean that on additional CMake runs the tag will not be overwritten, allowing you to benefit from a build cache